include/onnxruntime/core/framework/data_types.h

```diff
@@ -105,12 +107,16 @@ class DataTypeImpl {
     return false;
   }
 
+  virtual bool IsOptionalType() const {
+    return false;
+  }
+
   // Returns this if this is of tensor-type and null otherwise
   virtual const TensorTypeBase* AsTensorType() const {
     return nullptr;
   }
 
-  virtual const SequenceTensorTypeBase* AsSequenceTensorBase() const {
+  virtual const SequenceTensorTypeBase* AsSequenceTensorType() const {
```

Comment:
Consistency fix.

Reply:
Well, it is still returning Base. Perhaps, it would be simpler to rename AsSparseTensorType?

Reply:
Another thought. I know this is the ABI, but it is still in the include folder for ORT, so it is a public interface of sorts. Should we be more careful?

Reply:
If it isn't ABI, is there some value in making things consistent ? Personally (and maybe this isn't the same for others), I found the variation in naming patterns dampen my understanding...